Dataset analysis: https://www.kaggle.com/datasets/prince7489/online-vs-offline-learning-efficiency-dataset

In [1]:
%%sql
SELECT *
FROM online_vs_offline_learning_dataset
LIMIT 100;

,Learning_Mode,Subject,Study_Hours,Retention_Score,Focus_Level,Exam_Score
0,Offline,English,7.7,51,96,70
1,Offline,English,6.2,90,82,81
2,Online,English,1.2,75,66,71
3,Online,Math,6.5,80,64,77
4,Online,English,5.5,95,58,78
...,...,...,...,...,...,...
95,Online,Programming,5.6,73,54,69
96,Offline,Math,7.5,50,79,66
97,Online,English,6.2,95,79,89
98,Offline,Math,5.1,63,69,63


In [3]:
%%sql
SELECT
COUNT(*) AS Student_count
FROM online_vs_offline_learning_dataset


,Student_count
0,1000


Which method gives better results? // Która metoda wywołuje lepsze wyniki?

In [1]:
%%sql
SELECT
    Learning_Mode,
    ROUND(AVG(Exam_Score), 2) AS AVG_Score,
    ROUND(AVG(Retention_Score), 2) AS AVG_Retention_Score,
    COUNT(*) AS Student_count
FROM online_vs_offline_learning_dataset
GROUP BY Learning_Mode;

,Learning_Mode,AVG_Score,AVG_Retention_Score,Student_count
0,Online,70.21,70.66,500
1,Offline,73.14,74.72,500


Conclusion. he dataset is perfectly split with 500 students in each learning mode (online vs. offline). This makes the comparison fair and statistically reliable. The average exam score (AVG_Score) for the offline group (73.14) is about 3 points higher compared to the online group (70.21). Offline students also demonstrate better long-term knowledge retention (AVG_Retention_Score is 74.72 for offline versus 70.66 for online).

Baza danych jest idealnie podzielona mamy po 500 studentów uczących się online i offline. Dzięki temu porównanie jest sprawiedliwe i wysoce wiarygodne. Średni wynik z egzaminu (AVG_Score) dla grupy stacjonarnej (73.14) jest wyższy o około 3 punkty w porównaniu do grupy zdalnej (70.21). Studenci uczący się stacjonarnie wykazują również lepsze długoterminowe zapamiętywanie materiału (AVG_Retention_Score wynosi 74.72 dla offline względem 70.66 dla online).

Finding the best learners and the lucky ones // Znalezienie osób najlepiej uczących się i szczęściarzy

In [11]:
%%sql
SELECT
    Learning_Mode,
    Study_Hours,
    Exam_Score
FROM online_vs_offline_learning_dataset
WHERE Exam_Score > 90
ORDER BY Exam_Score DESC;

,Learning_Mode,Study_Hours,Exam_Score
0,Offline,7.0,100
1,Offline,6.5,100
2,Online,4.1,100
3,Offline,4.4,100
4,Online,5.4,100
...,...,...,...
80,Online,7.7,91
81,Online,1.5,91
82,Offline,2.7,91
83,Online,6.5,91


Among the 85 top-performing students, nearly two-thirds (65.9%) studied in person (Offline). Online students make up only 34.1% of this elite group. Despite the overall offline advantage, achieving a perfect score (100 points) is possible in both environments. The top of the table shows a mix of Offline and Online students hitting the maximum score. Students scoring above 90 studied anywhere from 1.1 to just 8 hours. Efficiency is highly individual here for instance, one Offline student scored 100 after only 1.3 hours of study, while another required 7 hours for the same result.

While outstanding results can be achieved regardless of the learning method, the Offline environment makes a student nearly twice as likely to be among the top performers. This suggests that in-person learning, with its direct engagement, better facilitates mastery of the material.

Wśród 85 najlepszych studentów, prawie dwie trzecie (65.9%) uczyło się stacjonarnie (Offline). Studenci uczący się zdalnie (Online) stanowią tylko 34.1% tej elitarnej grupy. Mimo przewagi trybu offline, osiągnięcie idealnego wyniku (100 punktów) jest możliwe w obu środowiskach. W czołówce tabeli widzimy zarówno studentów Offline, jak i Online. tudenci z wynikami powyżej 90 punktów uczyli się od 1.1 do zaledwie 8 godzin. Efektywność jest tu bardzo zróżnicowana – na przykład, jeden ze studentów Offline zdobył 100 punktów po zaledwie 1.3h nauki, podczas gdy inny potrzebował na to 7 godzin.

Chociaż wybitne wyniki można osiągnąć niezależnie od metody nauczania, to środowisko stacjonarne (Offline) niemal dwukrotnie zwiększa szansę na znalezienie się w grupie najlepszych studentów. Może to sugerować, że bezpośredni kontakt z materiałem i wykładowcą ułatwia opanowanie wiedzy na poziomie zaawansowanym.

In [16]:
%%sql
SELECT
    Learning_Mode,
    MIN(Study_Hours) AS min_hours,
    MAX(Study_Hours) AS max_hours,
    ROUND(AVG(Study_Hours), 2) AS avg_hours_the_best_person
FROM online_vs_offline_learning_dataset
WHERE Exam_Score > 90
GROUP BY Learning_Mode;

,Learning_Mode,min_hours,max_hours,avg_hours_the_best_person
0,Online,1.3,8.0,5.14
1,Offline,1.1,7.9,4.80


In [7]:
%%sql
SELECT
    Learning_Mode,
    Study_Hours,
    Exam_Score
FROM online_vs_offline_learning_dataset
WHERE Study_Hours < 3 AND Exam_Score > 80;

,Learning_Mode,Study_Hours,Exam_Score
0,Online,2.5,85
1,Offline,1.5,85
2,Offline,2.7,81
3,Offline,1.5,89
4,Offline,1.6,82
...,...,...,...
59,Online,2.5,81
60,Online,2.4,84
61,Offline,1.3,82
62,Online,2.1,87


Regardless of the learning mode, top performers can achieve great results studying anywhere from roughly 1 hour (1.1h for offline, 1.3h for online) up to a maximum of 8 hours.  There are 64 students who managed to secure a high score (over 80 points) with minimal study time (under 3 hours). The majority of these highly efficient students (54.7%) belong to the offline group.

In [9]:
%%sql
SELECT
    Learning_Mode,
    MIN(Study_Hours) AS min_hours,
    MAX(Study_Hours) AS max_hours,
    ROUND(AVG(Study_Hours), 2) AS avg_hours
FROM online_vs_offline_learning_dataset
WHERE Exam_Score >= 80
GROUP BY Learning_Mode;

,Learning_Mode,min_hours,max_hours,avg_hours
0,Online,1.0,8,4.67
1,Offline,1.1,8,4.65


Learning mode does not affect the study time required for good results.
To score 80 points or higher, students in both groups need to put in almost identical average independent study time (around 4.65 hours). The distribution of effort is practically the same across both modes (from 1 to 8 hours).

Tryb nauki nie ma wpływu na czas potrzebny do osiągnięcia dobrych wyników.
Aby zdobyć minimum 80 punktów na egzaminie, studenci z obu grup muszą poświęcić średnio niemal identyczną ilość czasu na własną naukę (około 4,65 godziny). Rozkład w obu grupach jest praktycznie taki sam (od 1 do 8 godzin).



Study effort and the result? // Wysiłek w nauce, a rezultat?

In [20]:
%%sql
SELECT
    Learning_Mode,
    CASE
    WHEN Study_Hours < 5 THEN 'Low <5 hours'
    WHEN Study_Hours BETWEEN 5 AND 10 THEN 'Medium 5-10 hoours'
    ELSE 'High > 10 hours'
END AS Study_Intensity,
ROUND(AVG(Exam_Score), 2) AS Avg_Score
FROM online_vs_offline_learning_dataset
GROUP BY Learning_Mode, Study_Intensity
ORDER BY Learning_Mode, Avg_Score DESC;

,Learning_Mode,Study_Intensity,Avg_Score
0,Offline,Low <5 hours,73.34
1,Offline,Medium 5-10 hoours,72.91
2,Online,Medium 5-10 hoours,71.35
3,Online,Low <5 hours,69.27


Offline is more optimal: Offline students achieve their highest average (73.34) with low study intensity (under 5 hours). Studying more (5-10h) does not improve their average score. This implies they absorb most of the required knowledge directly during classes.

Online requires more effort: In online mode, there is a clear correlation. Students studying less (<5h) have the lowest average score (69.27). To improve their results, they must put in medium effort (5-10h), which raises their average to 71.35.

Overall Offline Advantage: An analysis of 1000 evenly distributed students showed that the offline mode yields better results. Average exam scores are 3 points higher, and these students demonstrate noticeably better long-term knowledge retention.

Producing Top Performers: The in-person environment strongly favors outstanding academic achievements. Nearly 66% of all top-performing students (scoring > 90 points) belong to the Offline cohort.

Time Economy: Although achieving a "good" score (> 80 points) requires an identical average independent study time for both groups (~4.6 hours), Offline students are more time-efficient overall. They maintain high average scores with minimal independent effort (<5h), whereas Online students must increase their self-study hours to match the performance of the in-person group.

Ogólna przewaga trybu Offline: Analiza 1000 studentów (równo podzielonych) wykazała, że tryb stacjonarny przynosi lepsze efekty. Średnie wyniki z egzaminów są o 3 punkty wyższe, a studenci ci wykazują odczuwalnie wyższą zdolność długoterminowego zapamiętywania materiału (retencji).

Fabryka "Prymusów": Środowisko stacjonarne zdecydowanie sprzyja osiąganiu wybitnych wyników. Aż blisko 66% studentów z wynikami powyżej 90 punktów to osoby uczące się w trybie Offline.

Ekonomia czasu: Mimo że do osiągnięcia "dobrego" wyniku (powyżej 80 pkt) studenci z obu grup poświęcają średnio tyle samo czasu w domu (ok. 4,6 h), to studenci Offline są bardziej efektywni. Wymagają mniej wysiłku własnego (<5h), aby utrzymać wysoką średnią, podczas gdy studenci Online muszą systematycznie zwiększać czas nauki w domu, aby dorównać wynikom grupy stacjonarnej.